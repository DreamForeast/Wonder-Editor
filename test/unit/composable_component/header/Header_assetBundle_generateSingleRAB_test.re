open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

open Js.Promise;

let _ =
  describe("Header AssetBundle", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());

      MainEditorSceneTool.initInspectorEngineState(
        ~sandbox,
        ~isInitJob=false,
        ~noWorkerJobRecord=
          NoWorkerJobConfigToolEngine.buildNoWorkerJobConfig(
            ~initPipelines=
              {|
             [
              {
                "name": "default",
                "jobs": [
                    {"name": "init_inspector_engine" }
                ]
              }
            ]
             |},
            ~initJobs=
              {|
             [
                {"name": "init_inspector_engine" }
             ]
             |},
            (),
          ),
        (),
      );

      StateInspectorEngineService.unsafeGetState()
      |> MainUtils._handleInspectorEngineState
      |> StateInspectorEngineService.setState
      |> ignore;

      CanvasTool.prepareInspectorCanvasAndImgCanvas(sandbox) |> ignore;

      MainEditorSceneTool.prepareScene(sandbox);
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test generate single rab", () => {
      let truckWDBArrayBuffer = ref(Obj.magic(1));

      beforeAll(() =>
        truckWDBArrayBuffer := WDBTool.convertGLBToWDB("CesiumMilkTruck")
      );

      beforeEach(() => {
        MainEditorAssetTool.buildFakeFileReader();
        MainEditorAssetTool.buildFakeImage();

        LoadTool.buildFakeTextDecoder(LoadTool.convertUint8ArrayToBuffer);
        LoadTool.buildFakeURL(sandbox^);

        LoadTool.buildFakeLoadImage(.);
      });

      describe("test buildSelectTreeForGenerateSingleRAB", () => {
        testPromise("test1", () => {
          let addedMaterialNodeId1 = MainEditorAssetIdTool.getNewAssetId();
          MainEditorAssetHeaderOperateNodeTool.addMaterial();

          let addedFolderNodeId1 = MainEditorAssetIdTool.getNewAssetId();
          MainEditorAssetHeaderOperateNodeTool.addFolder();

          MainEditorAssetTreeTool.Select.selectFolderNode(
            ~nodeId=addedFolderNodeId1,
            (),
          );

          MainEditorAssetUploadTool.loadOneTexture(
            ~imgName="image1.png",
            ~imgSrc="newImgBase64",
            (),
          )
          |> then_(uploadedTextureNodeId =>
               MainEditorAssetUploadTool.loadOneWDB(
                 ~fileName="Truck",
                 ~arrayBuffer=truckWDBArrayBuffer^,
                 (),
               )
               |> then_(uploadedWDBNodeId =>
                    HeaderAssetBundleTool.GenerateSingleRAB.buildGenerateSingleRABModal(
                      ~selectTree=
                        HeaderAssetBundleTool.GenerateSingleRAB.buildSelectTreeForGenerateSingleRAB
                        |> StateLogicService.getStateToGetData,
                      ~send=SinonTool.createOneLengthStub(sandbox^),
                      (),
                    )
                    |> BuildComponentTool.buildUI
                    |> ReactTestTool.createSnapshotAndMatch
                    |> resolve
                  )
             );
        });
        test("test2", () => {
          MainEditorAssetHeaderOperateNodeTool.addFolder();

          MainEditorAssetHeaderOperateNodeTool.addScriptEventFunction();

          HeaderAssetBundleTool.GenerateSingleRAB.buildGenerateSingleRABModal(
            ~selectTree=
              HeaderAssetBundleTool.GenerateSingleRAB.buildSelectTreeForGenerateSingleRAB
              |> StateLogicService.getStateToGetData,
            ~send=SinonTool.createOneLengthStub(sandbox^),
            (),
          )
          |> BuildComponentTool.buildUI
          |> ReactTestTool.createSnapshotAndMatch;
        });
      });

      describe("build resource data", () =>
        describe(
          "build selected assets from selectTree to be resource data", () => {
          describe("test resource data->basicSourceTextureImageDataMap", () =>
            testPromise("test", () => {
              let imageBase64 = Base64Tool.buildFakeBase64_1();

              MainEditorAssetUploadTool.loadOneTexture(
                ~imgName="image1.png",
                ~imgSrc=imageBase64,
                (),
              )
              |> then_(uploadedTextureNodeId1 => {
                   let selectTree =
                     HeaderAssetBundleTool.GenerateSingleRAB.buildSelectTreeForGenerateSingleRAB
                     |> StateLogicService.getStateToGetData
                     |> SelectTreeTool.setSelectForSelectTree(
                          true,
                          MainEditorAssetTextureNodeTool.getTextureName(
                            ~nodeId=uploadedTextureNodeId1,
                            (),
                          ),
                        );

                   let (
                     basicMaterials,
                     lightMaterials,
                     basicSourceTextures,
                     cubemapTextures,
                     geometrys,
                     scriptEventFunctionDataArr,
                     scriptAttributeDataArr,
                     basicSourceTextureImageDataMap,
                   ) =
                     HeaderAssetBundleTool.GenerateSingleRAB.generateSingleRABResourceData(
                       selectTree,
                     )
                     |> StateLogicService.getStateToGetData;

                   let editorState = StateEditorService.getState();
                   let imageDataIndex1 =
                     MainEditorAssetTextureNodeTool.getTextureImageDataIndex(
                       uploadedTextureNodeId1,
                       editorState,
                     );
                   let {uint8Array, name, mimeType}: ImageDataType.imageData =
                     BasicSourceTextureImageDataMapAssetEditorService.unsafeGetData(
                       imageDataIndex1,
                       editorState,
                     );
                   basicSourceTextureImageDataMap
                   |> expect
                   == (
                        WonderCommonlib.ImmutableSparseMapService.createEmpty()
                        |> WonderCommonlib.ImmutableSparseMapService.set(
                             imageDataIndex1,
                             {
                               uint8Array:
                                 BufferUtils.convertBase64ToUint8Array(
                                   imageBase64,
                                 ),
                               name,
                               mimeType,
                             }: HeaderAssetBundleType.imageData,
                           )
                      )
                   |> resolve;
                 });
            })
          );

          describe("test resource data->material data", () => {
            describe(
              "add light material contained textureData to textures", () =>
              testPromise(
                "if selectTree->textures not has selectTree->lightMaterials contained textureData, resourceData->textures should has them",
                () => {
                  let addedMaterialNodeId1 =
                    MainEditorAssetIdTool.getNewAssetId();
                  MainEditorAssetHeaderOperateNodeTool.addMaterial();

                  let addedMaterialNodeId2 =
                    MainEditorAssetIdTool.getNewAssetId();
                  MainEditorAssetHeaderOperateNodeTool.addMaterial();

                  let addedFolderNodeId1 =
                    MainEditorAssetIdTool.getNewAssetId();
                  MainEditorAssetHeaderOperateNodeTool.addFolder();

                  MainEditorAssetTreeTool.Select.selectFolderNode(
                    ~nodeId=addedFolderNodeId1,
                    (),
                  );

                  MainEditorAssetUploadTool.loadOneTexture(
                    ~imgName="image1.png",
                    ~imgSrc=Base64Tool.buildFakeBase64_1(),
                    (),
                  )
                  |> then_(uploadedTextureNodeId1 =>
                       MainEditorAssetUploadTool.loadOneTexture(
                         ~imgName="image2.png",
                         ~imgSrc=Base64Tool.buildFakeBase64_2(),
                         (),
                       )
                       |> then_(uploadedTextureNodeId2 => {
                            MainEditorLightMaterialForAssetTool.dragAssetTextureToMap(
                              ~currentNodeId=addedMaterialNodeId2,
                              ~textureNodeId=uploadedTextureNodeId2,
                              ~material=
                                MainEditorAssetMaterialNodeTool.getMaterialComponent(
                                  ~nodeId=addedMaterialNodeId2,
                                  (),
                                ),
                              (),
                            );

                            let selectTree =
                              HeaderAssetBundleTool.GenerateSingleRAB.buildSelectTreeForGenerateSingleRAB
                              |> StateLogicService.getStateToGetData
                              |> SelectTreeTool.setSelectForSelectTree(
                                   true,
                                   MainEditorAssetMaterialNodeTool.getMaterialName(
                                     ~nodeId=addedMaterialNodeId2,
                                     (),
                                   ),
                                 )
                              |> SelectTreeTool.setSelectForSelectTree(
                                   true,
                                   MainEditorAssetTextureNodeTool.getTextureName(
                                     ~nodeId=uploadedTextureNodeId1,
                                     (),
                                   ),
                                 );

                            let (
                              basicMaterials,
                              lightMaterials,
                              basicSourceTextures,
                              cubemapTextures,
                              geometrys,
                              scriptEventFunctionDataArr,
                              scriptAttributeDataArr,
                              basicSourceTextureImageDataMap,
                            ) =
                              HeaderAssetBundleTool.GenerateSingleRAB.generateSingleRABResourceData(
                                selectTree,
                              )
                              |> StateLogicService.getStateToGetData;

                            let editorState = StateEditorService.getState();
                            (lightMaterials, basicSourceTextures)
                            |> expect
                            == (
                                 [|
                                   MainEditorAssetMaterialNodeTool.getMaterialComponent(
                                     ~nodeId=addedMaterialNodeId2,
                                     (),
                                   ),
                                 |],
                                 [|
                                   HeaderAssetBundleTool.GenerateSingleRAB.buildTextureData(
                                     MainEditorAssetTextureNodeTool.getTextureComponent(
                                       uploadedTextureNodeId1,
                                       editorState,
                                     ),
                                     MainEditorAssetTextureNodeTool.getTextureImageDataIndex(
                                       uploadedTextureNodeId1,
                                       editorState,
                                     ),
                                   ),
                                   HeaderAssetBundleTool.GenerateSingleRAB.buildTextureData(
                                     MainEditorAssetTextureNodeTool.getTextureComponent(
                                       uploadedTextureNodeId2,
                                       editorState,
                                     ),
                                     MainEditorAssetTextureNodeTool.getTextureImageDataIndex(
                                       uploadedTextureNodeId2,
                                       editorState,
                                     ),
                                   ),
                                 |],
                               )
                            |> resolve;
                          })
                     );
                },
              )
            );

            describe("fix bug", () =>
              test(
                "added light material shouldn't add to basic material resource data",
                () => {
                let addedMaterialNodeId1 =
                  MainEditorAssetIdTool.getNewAssetId();
                MainEditorAssetHeaderOperateNodeTool.addMaterial();

                let selectTree =
                  HeaderAssetBundleTool.GenerateSingleRAB.buildSelectTreeForGenerateSingleRAB
                  |> StateLogicService.getStateToGetData
                  |> SelectTreeTool.setSelectForSelectTree(
                       true,
                       MainEditorAssetMaterialNodeTool.getMaterialName(
                         ~nodeId=addedMaterialNodeId1,
                         (),
                       ),
                     );

                let (
                  basicMaterials,
                  lightMaterials,
                  basicSourceTextures,
                  cubemapTextures,
                  geometrys,
                  scriptEventFunctionDataArr,
                  scriptAttributeDataArr,
                  basicSourceTextureImageDataMap,
                ) =
                  HeaderAssetBundleTool.GenerateSingleRAB.generateSingleRABResourceData(
                    selectTree,
                  )
                  |> StateLogicService.getStateToGetData;

                basicMaterials |> expect == [||];
              })
            );
          });

          describe("test resource data->geometrys", () =>
            testPromise("test", () =>
              MainEditorAssetUploadTool.loadOneWDB(
                ~fileName="Truck",
                ~arrayBuffer=truckWDBArrayBuffer^,
                (),
              )
              |> then_(uploadedWDBNodeId => {
                   let selectTree =
                     HeaderAssetBundleTool.GenerateSingleRAB.buildSelectTreeForGenerateSingleRAB
                     |> StateLogicService.getStateToGetData
                     |> SelectTreeTool.setSelectForSelectTree(
                          true,
                          HeaderAssetBundleTool.GenerateSingleRAB.buildWDBGeometryFolderName(
                            MainEditorAssetWDBNodeTool.getWDBName(
                              ~nodeId=uploadedWDBNodeId,
                              (),
                            ),
                          ),
                        );

                   let (
                     basicMaterials,
                     lightMaterials,
                     basicSourceTextures,
                     cubemapTextures,
                     geometrys,
                     scriptEventFunctionDataArr,
                     scriptAttributeDataArr,
                     basicSourceTextureImageDataMap,
                   ) =
                     HeaderAssetBundleTool.GenerateSingleRAB.generateSingleRABResourceData(
                       selectTree,
                     )
                     |> StateLogicService.getStateToGetData;

                   geometrys |> expect == [|3, 4, 5, 2|] |> resolve;
                 })
            )
          );
        })
      );
    });
  });