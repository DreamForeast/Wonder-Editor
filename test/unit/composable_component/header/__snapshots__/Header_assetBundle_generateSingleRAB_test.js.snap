// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header AssetBundle test generate single rab test buildSelectTreeForGenerateSingleRAB test1 1`] = `
<article
  className="wonder-singleInput-modal"
>
  <div
    className="modal-item"
  >
    <div
      className="modal-item-header"
    >
      Generate Single RAB
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
    <div
      className="modal-item-content"
    >
      <div
        className="content-field"
      >
        <div
          className="field-title"
        >
          name
        </div>
        <div
          className="field-content"
        >
          <input
            className="input-component"
            onChange={[Function]}
            type="text"
            value="WonderSingleRAB"
          />
        </div>
      </div>
      <article
        className="wonder-selectTree"
      >
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="tree-container"
            >
              <input
                defaultChecked={false}
                onClick={[Function]}
                type="checkbox"
              />
              <img
                className="treeNode-icon"
                src="./public/img/selectFolder.png"
              />
              Assets
            </div>
          </li>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="tree-container"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
                <img
                  className="treeNode-icon"
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAGb0lEQVRoQ+2azUsVfRTHj5pvaaGIitTChW5DUHDhC6KCighmGuomwo2I/4aQ4kLFCNuEiGBgtBBRcaGgiLRw4V4oUTDFgohKTX34Ds+5nHvu+c2duRo8xDMwzL3jzL3nc7/n7XfGpK6urmv6C7ak/0H+YyreqiIPHz6k2tpaKi4upuTkZEpKSvJ2vX38+JHW19dpf3//1n6OG4PA+CdPnlBWVpZntAZwwYDg+vqafvz4QW/fvqVPnz7dCCphkPv371N/f79nOO9stDzCOq0K3gOCN7y+urqiN2/eJKxSaJC0tDQaHBykO3fuhIKwXEyC4DVgsA8PD9PZ2VkohUKBlJWVUUNDA6WkpESpIN2JDZYuJc9Z1rE6OLI6S0tL9OHDh8AwgUEaGxvp0aNHURAawM+1XIpICI4bhtnZ2aGFhYVAMIFAoEJYCAtSW8QK3AZMXJCKigqqqqrylNAupRWQxsskwOclCP/q8igVkW62trZGGxsbvsr4ghQVFVFnZ2dMYFu/Np9jAAbX8NJADu7Ly0svyC2F+JqpqSk6Pj52wviCDAwMeBBSDV0r5HuGwD1y5/s57UqA379/E3YLRkLj70NDQ+FBnj17Rvfu3XO6lMuNYHRqamrMjvMc8ACB8RcXF5E9CAyuf/HihQljKlJYWEiPHz/2jPFLtZY7QQncl56eTqg5fLRAzs/PvXqBI0OxWtLNOI7wt+npaTo4OIiBMUH6+vo8I9itZOXW8SGDmiEYICMjg7DjvLUB4NevX97OMFCG4wVHTgAMiL+jYOotBqSgoIDa29vjxkY8NQCQmZnpgbg2GAeInz9/RpTxczGGefnyJX379i3qY2NAuru7CX2UVkMaLtOuDnCowUoABO/9NgaRqrgCn0HghqOjo/4gz58/93w8rFtxkLNbASIsCOKFs5isLzJeAAPQkZERN0h+fj61tbVFUqdV1HT6lXWDs5VUBMHu157ArThOGCRIKp6dnaXDw8MITJRr1dTUUGlpqQli1Q8+BxgoIjMWAKAIjgD0C3YAcPYKkoahyt7eHr17984GQXzgy7mY6bWGlbEYhosejIZ7sYtx+sVnsjJwFfi5TL86a+lKL90LiuH68fFxG6S3t9f7Ba1AD9qWsCoSSNYjXnfogggwjg8LQqZhgODasbExN4hVP6wVoKUOq8Iwuk3BPQzCxrDx7FI4r4uhbCZlPXGC9PT0mIUwKIgMfFZVqitdi7MPDJe7hrAWXXy907UQI4kqIgOfg18eZSsvWw7+hXVrolt67VqAmZiYsF3r6dOnCcWItS7RiUJPU2Rn61qTaBi+h91ycnLSBsHag4NdDxeCLGvZWN0Zu0ZCrlhgAAuElUPWevXqlQ2CHgvpV2YZ7n5ddcRap+tz+DZLEbbCWuq6YDi2UHdev35tg5SXl1NJSUmkRbEqu58y0uBEpijSeJdbMcju7m7U8jeqsmMh1dzcHAVirdMtGA1hqeA327IgLBjOWHNzc/T161dbEZzt6OiIrO4khJWCXW7FENZRtioyHcvzOuXqqo6aI93K+x79WKG+vp7y8vJiVod+iyvtRnogJ5XwU0WPUXUhxN8BcXp6SvPz85I9FgSda0tLS6SeyCLnN3jQruQHwxaEUURWdCx30dJEqWs96EErj8Loyl5+GSyeW4VRRNcaxAem92jh9Wau2aFKU1OTByNHQTxAiJeK48HEWCFOWPHBTSRUmJmZ8dwrEAguqquro5ycHA/GajV00XNlLQmlXYrfB4kNqIEs9f79e/N38B3Qtba2epVeTw2tVaKMCeu1C0LDMJScpPD6BWq4Nl+Q3Nxcb+6rXcwaRHCL7spgljIuCD1WhSutrKzQyclJYiC4q7KykjAiCrpq9ANxwbjqBlfxz58/0+rqql9oxaZf6+rq6movXvQc2DUicqXiqHT570PSeBBfvnyh5eVlXwjv+4I+Z0cfhlEqp2TOYK6KHwQG17AbydesBJ7+bm9vx4UIBYKLMS4CEC9l41V7DWNZZM14ERObm5t0dHQUCCI0CH8qUvPdu3edz02sVOyySPdRmHMtLi4GBohkw6CupT8ZGQ1Ps+Bqss5YjWQ8CHYluBECO5EtcIy4Pjw7O9sDkvXGBaMLHwC+f/9OW1tb3rTxJtuNQeSXQyU8rnvw4EHUf0Do5hC+j0CW64mbQCQcIzf90j9x/60q8icMDPqZfw3IP2F1JSnkGOAPAAAAAElFTkSuQmCC"
                />
                New Material
              </div>
            </li>
          </ul>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="tree-container"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
                <img
                  className="treeNode-icon"
                  src="./public/img/selectFolder.png"
                />
                New Folder
              </div>
            </li>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="tree-container"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                  <img
                    className="treeNode-icon"
                    src="newImgBase64"
                  />
                  image1
                </div>
              </li>
            </ul>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="tree-container"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                  <img
                    className="treeNode-icon"
                    src="./public/img/selectFolder.png"
                  />
                  Truck_Geometrys
                </div>
              </li>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="./public/img/geo.png"
                    />
                    Cesium_Milk_Truck_0
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="./public/img/geo.png"
                    />
                    Cesium_Milk_Truck_1
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="./public/img/geo.png"
                    />
                    Cesium_Milk_Truck_2
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="./public/img/geo.png"
                    />
                    Wheels
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="./public/img/geo.png"
                    />
                    Wheels
                  </div>
                </li>
              </ul>
            </ul>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="tree-container"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                  <img
                    className="treeNode-icon"
                    src="./public/img/selectFolder.png"
                  />
                  Materials
                </div>
              </li>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAEBAQEBAQEBAQE"
                    />
                    truck
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAEBAQEBAQEBAQE"
                    />
                    glass
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAEBAQEBAQEBAQE"
                    />
                    window_trim
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAEBAQEBAQEBAQE"
                    />
                    wheels
                  </div>
                </li>
              </ul>
            </ul>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="tree-container"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                  <img
                    className="treeNode-icon"
                    src="./public/img/selectFolder.png"
                  />
                  Textures
                </div>
              </li>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="./public/null.png"
                    />
                    basicSourceTexture_0
                  </div>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="tree-container"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                    <img
                      className="treeNode-icon"
                      src="./public/null.png"
                    />
                    basicSourceTexture_1
                  </div>
                </li>
              </ul>
            </ul>
          </ul>
        </ul>
      </article>
    </div>
    <div
      className="modal-item-footer"
    >
      <button
        className="footer-submit"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
</article>
`;

exports[`Header AssetBundle test generate single rab test buildSelectTreeForGenerateSingleRAB test2 1`] = `
<article
  className="wonder-singleInput-modal"
>
  <div
    className="modal-item"
  >
    <div
      className="modal-item-header"
    >
      Generate Single RAB
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
    <div
      className="modal-item-content"
    >
      <div
        className="content-field"
      >
        <div
          className="field-title"
        >
          name
        </div>
        <div
          className="field-content"
        >
          <input
            className="input-component"
            onChange={[Function]}
            type="text"
            value="WonderSingleRAB"
          />
        </div>
      </div>
      <article
        className="wonder-selectTree"
      >
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="tree-container"
            >
              <input
                defaultChecked={false}
                onClick={[Function]}
                type="checkbox"
              />
              <img
                className="treeNode-icon"
                src="./public/img/selectFolder.png"
              />
              Assets
            </div>
          </li>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="tree-container"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
                <img
                  className="treeNode-icon"
                  src="./public/img/selectFolder.png"
                />
                New Folder
              </div>
            </li>
          </ul>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="tree-container"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
                <img
                  className="treeNode-icon"
                  src="./public/img/selectJsFunc.png"
                />
                New Script EventFunction
              </div>
            </li>
          </ul>
        </ul>
      </article>
    </div>
    <div
      className="modal-item-footer"
    >
      <button
        className="footer-submit"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
</article>
`;
